src/runtime_src/core/common/pskernel_parse.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#define XRT_CORE_COMMON_SOURCE
+#include <regex>
+#include "pskernel_parse.h"
+
+
+namespace xrt_core { namespace pskernel {
+
+// Function to match function name
+// and extract formal parameters
+std::vector<xrt_core::pskernel::kernel_argument>
+extract_args (Dwarf_Die *die)
+{
+  Dwarf_Die child;
+  Dwarf_Attribute attr_mem;
+  Dwarf_Die type_mem;
+  Dwarf_Die *type = NULL;
+  Dwarf_Word var_size;
+  const char *var_name;
+  std::vector<xrt_core::pskernel::kernel_argument> return_args;
+
+  // Mapping for FFI types
+  static const std::map<std::pair<Dwarf_Word, Dwarf_Word>, ffi_type*> typeTable = {
+    std::make_pair(std::make_pair(DW_ATE_unsigned_char, 1), &ffi_type_uint8),
```

Comment:
Modern C++ is like Python, just faster :-)
```suggestion
    { { DW_ATE_unsigned_char, 1 }, &ffi_type_uint8 },
```
and so on.
https://en.cppreference.com/w/cpp/container/map